typing/typecore.ml

```diff
@@ -5814,6 +5886,9 @@ and type_let
     | Pexp_fun _ | Pexp_function _ -> true
     | Pexp_constraint (e, _)
     | Pexp_newtype (_, e)
+    | Pexp_apply
+      ({ pexp_desc = Pexp_extension({txt = "extension.unique"}, PStr []) },
+        [Nolabel, e])
```

Comment:
This is a pre-existing issue, but probably these cases should work for `ocaml.unique`, `unique`, `local` and `ocaml.local` too.